packages/transformer/src/IModelExporter.ts

```diff
@@ -220,26 +239,54 @@ export class IModelExporter {
   private _excludedElementCategoryIds = new Set<Id64String>();
   /** The set of classes of Elements that will be excluded (polymorphically) from transformation to the target iModel. */
   private _excludedElementClasses = new Set<typeof Element>();
-  /** The set of classes of ElementAspects that will be excluded (polymorphically) from transformation to the target iModel. */
-  private _excludedElementAspectClasses = new Set<typeof ElementAspect>();
-  /** The set of classFullNames for ElementAspects that will be excluded from transformation to the target iModel. */
-  private _excludedElementAspectClassFullNames = new Set<string>();
   /** The set of classes of Relationships that will be excluded (polymorphically) from transformation to the target iModel. */
   private _excludedRelationshipClasses = new Set<typeof Relationship>();
 
+  /** Strategy for how ElementAspects are exported */
+  private _exportElementAspectsStrategy: ExportElementAspectsStrategy<ElementAspectsHandler>;
+
   /** Construct a new IModelExporter
    * @param sourceDb The source IModelDb
    * @see registerHandler
    */
-  public constructor(sourceDb: IModelDb) {
+  public constructor(sourceDb: IModelDb, elementAspectsStrategy: ElementAspectExportStrategy = ElementAspectExportStrategy.WithElement) {
     this.sourceDb = sourceDb;
+    this._exportElementAspectsStrategy = this.createExportElementAspectsStrategy(sourceDb, elementAspectsStrategy);
+  }
+
+  private createExportElementAspectsStrategy(sourceDb: IModelDb, elementAspectsStrategy: ElementAspectExportStrategy): ExportElementAspectsStrategy<ElementAspectsHandler> {
+    switch(elementAspectsStrategy) {
```

Comment:
instead of an enum, why not just take the actual class to call new on? (as an argument to this and the constructor)
```ts
// contrived, probably needs adjustments
interface ElementAspectExportStrategyHandler {
  trackProgress: () => Promise<void>, // ew :/ can't this be made sync?
  onExportElementMultiAspects: (multiAspects) => Id64String[] | void,
  onExportElementUniqueAspect: (uniqueAspect, isUpdate) => Id64String | void,
  shouldExportElementAspect: (aspect: ElementAspect) => boolean,
}
//...

private createExportElementAspectsStrategy(
  sourceDb: IModelDb,
  elementAspectsStrategy: new (
    src: IModelDb,
    handler: ElementAspectExportStrategyHandler
  ) => ExportElementAspectsStrategy
) {
  return new HandlerClass(src, {
    trackProgress: async () => this.trackProgress(),
    onExportElementMultiAspects: (multiAspects) => this.handler.onExportElementMultiAspects(multiAspects),
    onExportElementUniqueAspect: (uniqueAspect, isUpdate) => this.handler.onExportElementUniqueAspect(uniqueAspect, isUpdate),
    shouldExportElementAspect: (aspect) => this.handler.shouldExportElementAspect(aspect),
  });
}

```

Reply:
Yeah, enum was unnecessary here. However, instead of passing constructor I opted to just pass the created strategy object itself. One of my intentions was to allow strategies to support different handler interfaces so that it would be future proof with contract changes. That way there was an option to change aspect exporting methods in `IModelExportHandler` and leave the old strategies unaffected until they are removed. This would allow experimenting with exporting ElementMultiAspects one by one or exporting ElementUniqueAspects and ElementMultiAspects through one method like `shouldExportElementAspect` works with both or different interface changes. So I moved handler creation to strategy itself where each strategy can map handler methods correctly.